manifests/charts/ztunnel/templates/daemonset.yaml

```diff
@@ -4,7 +4,10 @@ metadata:
   name: ztunnel
   namespace: {{ .Release.Namespace }}
   labels:
-    {{- .Values.labels | toYaml | nindent 4}}
+    {{ with .Values.labels -}}{{ toYaml . | indent 4 }}{{ end }}
+    install.operator.istio.io/owning-resource: {{ .Values.ownerName | default "unknown" }}
```

Comment:
Why do we need this? seems like we have plenty of objects we don't do this for.

I really do not like putting junk for istioctl in the helm charts. if it wants to inject junk it can do so in the istioctl code?

Reply:
> Why do we need this? seems like we have plenty of objects we don't do this for.

In istioctl, these labels are automatically added during the installation process so the uninstallation behavior is alright. However if installed using helm manifests, every component has these labels will be pruned by istioctl except objects without these labels, like ztunnel.

> I really do not like putting junk for istioctl in the helm charts. if it wants to inject junk it can do so in the istioctl code?

I don't like either, but I'm not sure if we need to or how we should handle the inconsistency pruning behavior of cni and ztunnel.

Reply:
I found this was completed in https://github.com/istio/istio/pull/28039. I guess since then, no similar changes have been made, resulting in the current inconsistency.